### PR TITLE
Introduce top-level test directory and copy rh-operators e2e tests

### PR DESCRIPTION
**Pull request by timflannagan**
Creates a top-level tests directory and copies the upstream OLM test/rh-operators directory test packages to ./test/rh-operators. Moving these test packages downstream will allow us to remove those tests upstream.

Updates the root Makefile and adds an e2e/ocp target that only runs those rh-operators e2e tests currently. In the future, as we continue moving content downstream, this will likely house any top-level downstream-only e2e tests.

### CODE REVIEW

File: test/rh-operators/metric_test.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	kubeconfig = os.Getenv("KUBECONFIG")
+	testlog    = logrus.New()
+)
+
+// TestRHOperators installs all Red Hat Operators available on the cluster and see that they install successfully.
```
Comment (njhale) [inline test/rh-operators/metric_test.go:25]: I've always been dubious about the value this test adds b/c:

- It takes a dependency on content we don't own, the redhat-operators Catalog; its correctness directly affects whether our tests pass
- That external content can change irrespective of changes to OLM, making test outcomes become non-deterministic within the scope of this repository
- OLM should already have more generic e2e tests that catch regressions that would preclude previously installable content

Simply put, I think we should seriously consider dropping this test -- and the overhead that comes with maintaining it -- entirely.

File: Makefile [hunk 2]
```diff
@@ -124,6 +124,9 @@ e2e/operator-registry: ## Run e2e registry tests
 e2e/olm: ## Run e2e olm tests
 	$(MAKE) e2e WHAT=operator-lifecycle-manager
 
+e2e/ocp:
+	go test -v -mod=vendor -timeout 70m ./test/...
```
[the remark this answers is not in the document]
Comment (timflannagan) [reply Makefile:128]: I had that same thought locally and ended up separating out the Makefile targets:

- This target more-or-less resembles something like `e2e/metrics` but it could instead be generalized upfront so we can easily iterate (without having to also update o/release)
- The OLM testing suite is long and prone to errors/flakes/etc.

**Comment by timflannagan** [comment]
Closing - likely heading in another direction.
